lib/reek/spec/smell_matcher.rb

```diff
@@ -0,0 +1,57 @@
+module Reek
+  module Spec
+    #
+    # Matches a +SmellWarning+ object agains a smell type and hash of attributes.
+    #
+    class SmellMatcher
+      attr_reader :smell_warning
+
+      COMPARABLE_ATTRIBUTES = %i(message lines context source)
+
+      def initialize(smell_warning)
+        @smell_warning = smell_warning
+      end
+
+      def matches?(klass, attributes = {})
+        matches_smell_type?(klass) && matches_attributes?(attributes)
+      end
+
+      def matches_smell_type?(klass)
+        smell_classes.include?(klass.to_s)
+      end
+
+      def matches_attributes?(attributes)
+        check_attributes_comparability(attributes)
+
+        main_attributes = attributes.slice(*COMPARABLE_ATTRIBUTES)
+        parameter_attributes = attributes.except(*COMPARABLE_ATTRIBUTES)
```

Comment:
> Y U all hatin' on activesupport???

Because it monkey-patches core classes.

It’s also very popular and many projects use antiquated versions (matching the rest of their Rails stack), so if we were depending on its newest version then Reek couldn’t be used with, say, Rails 3 projects (which I just upgraded from Rails 2.3!).